### PR TITLE
feat: [PIE-5731]: Add spot connector in cloud providers list

### PR DESCRIPTION
**Pull request by depanshu-dhiman**
### Summary
BE is sending the new Spot connector and by default, it's shown on the UI.
Clicking on it shows a blank popup which isn't ideal.
The change will put the Spot connector behind FF and it will hide it.

#### Screenshots
Screenshot 2022-10-17 at 6 49 35 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by depanshu-dhiman** [comment]
retrigger titlecheck

**Review by udayvunnam — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

File: src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts [hunk 1]
```diff
@@ -2197,6 +2197,8 @@ export const getConnectorDisplayName = (type: string): string => {
       return 'Custom Secrets Manager'
     case Connectors.GcpSecretManager:
       return 'GCP Secrets Manager'
+    case Connectors.SPOT:
+      return 'Spot'
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts:2201]: Please use constants like above `Connectors.GcpSecretManager`

**Review by sunnykesh-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts (hunk 1 above)
Comment (depanshu-dhiman) [reply src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts:2201]: Thanks. Will fix in following PRs.

Comment (depanshu-dhiman) [reply src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts:2201]: Wait... this is already a constant. Let's connect.

**Pull request merged**